src/main/java/org/folio/circulation/domain/validation/RequestQueueValidation.java

```diff
@@ -0,0 +1,78 @@
+package org.folio.circulation.domain.validation;
+
+import java.util.Objects;
+import java.util.function.Predicate;
+
+import org.folio.circulation.domain.Request;
+import org.folio.circulation.domain.reorder.ReorderRequest;
+import org.folio.circulation.resources.context.ReorderRequestContext;
+import org.folio.circulation.support.BadRequestFailure;
+import org.folio.circulation.support.RecordNotFoundFailure;
+import org.folio.circulation.support.Result;
+import org.folio.circulation.support.request.RequestHelper;
+
+public class RequestQueueValidation {
+
+  private RequestQueueValidation() {
+    throw new UnsupportedOperationException("Do not instantiate");
+  }
+
+  public static Result<ReorderRequestContext> queueFoundForItem(
+    Result<ReorderRequestContext> result) {
+    return result.failWhen(
+      r -> Result.succeeded(r.getRequestQueue().getRequests().isEmpty()),
+      r -> new RecordNotFoundFailure("Item", r.getItemId())
+    );
+  }
+
+  public static Result<ReorderRequestContext> queueIsConsistent(Result<ReorderRequestContext> result) {
```

Comment:
What is meant by consistent here, that the requests in the queue before and after the reorder are the same?